### ISSUE TITLE
Do not warn of unused import after E0433

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#48723]
Currently, if an import fails to resolve, we also emit a warning. The warning should be silenced.
```
error[E0433]: failed to resolve. Could not find `ext` in `os`
  --> src/lib.rs:37:22
   |
37 |         use std::os::ext::ffi::OsStringExt;
   |                      ^^^ Could not find `ext` in `os`

warning: unused import: `std::os::ext::ffi::OsStringExt`
   --> src/lib.rs:37:13
    |
37 |         use std::os::ext::ffi::OsStringExt;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default
```

_[Original report](https://internals.rust-lang.org/t/compiler-diagnostics-improvement-wishlist/6886/24)_

**Comment by zackmdavis** [comment]
(duplicate of #48244)
